mod-source-record-manager-server/src/main/java/org/folio/verticle/periodic/PeriodicDeleteJobExecutionVerticle.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.verticle.periodic;
+
+import lombok.extern.log4j.Log4j2;
+import org.folio.dao.JobExecutionDao;
+import org.folio.services.TenantDataProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Component;
+
+@Log4j2
+@Component
+@PropertySource("classpath:application.properties")
+public class PeriodicDeleteJobExecutionVerticle extends AbstractPeriodicJobVerticle {
+  /**
+   * Default unit time for job execution permanent delete interval = 24 hours.
+   */
+  @Value("${periodic.job.execution.permanent.delete.interval.ms:86400000}")
+  private long permanentDeletionInterval;
+  /**
+   * Default unit time for comparing marked as deleted jobs available to hard delete = 2 days.
+   */
+  @Value("${job.execution.difference.number.of.days:2}")
+  private long diffNumberOfDays;
+
+  @Autowired
+  private JobExecutionDao jobExecutionDao;
+  @Autowired
+  private TenantDataProvider tenantDataProvider;
+
+  private long timerId;
+
+  @Override
+  public void startPeriodicJob() {
+    timerId = vertx.setPeriodic(permanentDeletionInterval, handler -> proceedForJobExecutionPermanentDeletion());
```

Comment:
Thinking this call to `setPeriodic` would be in `AbstractPeriodicJobVerticle`. As it stands there is nothing that makes `AbstractPeriodicJobVerticle` "periodic". It is possible to implement a non-periodic verticle.

Reply:
Refactored abstract class, moved all periodic related code to abstract class

Reply:
looks great!